llama_index/indices/base.py

```diff
@@ -189,23 +189,43 @@ def insert(self, document: Document, **insert_kwargs: Any) -> None:
 
     @abstractmethod
     def _delete(self, doc_id: str, **delete_kwargs: Any) -> None:
-        """Delete a document."""
+        """Delete a node."""
 
-    def delete(self, doc_id: str, **delete_kwargs: Any) -> None:
-        """Delete a document from the index.
-
-        All nodes in the index related to the index will be deleted.
+    def delete(
+        self, doc_id: str, delete_from_docstore: bool = False, **delete_kwargs: Any
+    ) -> None:
+        """Delete a node from the index.
```

Comment:
this changes the existing behavior of delete, i wonder if we can mark `delete` as deprecated (with a logger warning), have it call `delete_ref_doc`, and also define `delete_nodes`

Reply:
Not quite. Delete still expects a node doc_id (previously, all the _delete() functions seemed to either expect a node doc_id or a ref doc_id, now they are all aligned to use node doc_ids and delete nodes

Reply:
We can definitely deprecate in favor of more explicit `delete_ref_doc` and `delete_nodes` though